digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/FloatingChecker.scala

```diff
@@ -16,8 +16,8 @@ class FloatingChecker(roadLinkService: RoadLinkService) {
     def outsideOfGeometry(ra: RoadAddress, roadLinks: Seq[RoadLinkLike]): Boolean = {
       !roadLinks.exists(rl => GeometryUtils.geometryLength(rl.geometry) > ra.startMValue) ||
         !roadLinks.exists(rl => GeometryUtils.areAdjacent(
-          GeometryUtils.truncateGeometry3D(rl.geometry, ra.startMValue, ra.endMValue),
-          ra.geometry))
+          GeometryUtils.truncateGeometry2D(rl.geometry, ra.startMValue, ra.endMValue),
```

Comment:
i got really bad feeling about 3D->2D change.  Is there any other way to fix this issue?

Reply:
is the problem in data (missing Z) or why do we have to convert 3D->2D?

Reply:
i mean if that is the case, we probably want to talk with JanneG about that. FTA might want to know if there is problem with the data

Reply:
i think Guilherme did this decause of the adjacency check. And since its comparing roadLinks[with N Point[x,y, and possibly z] with roadAddress endPoints [with only  N Point[x,y]) we wanted to ensure that z is always 0.0 for roadlink Points and z always 0.0 for roadAddress Points.
Still the *GeometryUtils.distance2DTo* compares adjacency by x and y only so in my opinion both the *truncateGeometry2D* and *truncateGeometry3D* could be used and also the *ra.geometry.map(_.copy(z = 0.0))* could be removed

Reply:
As @Campanha pointed out it is exactly that, also adding to it, the geometry we save on our own Road_Address table is already 2D truncated, therefore it was a way to keep the playing field in the same level.